core/types/src/firewall.rs

```diff
@@ -0,0 +1,66 @@
+use std::net::IpAddr;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct FirewallConfig {
+    pub connection_policy: ConnectionPolicyConfig,
+    pub rate_limiting: RateLimitingConfig,
+}
+
+// todo stricter defaults?
```

Comment:
When in doubt, I tend to avoid the default and let the users decide so that there are no suprises. Makes debugging easier imo.